runtime/binding-mqtt-kafka/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/kafka/internal/stream/MqttKafkaSubscribeFactory.java

```diff
@@ -166,21 +167,26 @@ public MessageConsumer newStream(
         final long routedId = begin.routedId();
         final long initialId = begin.streamId();
         final long authorization = begin.authorization();
+        final OctetsFW extension = begin.extension();
+        final MqttBeginExFW mqttBeginEx = extension.get(mqttBeginExRO::tryWrap);
 
-        final MqttKafkaBindingConfig binding = supplyBinding.apply(routedId);
+        assert mqttBeginEx.kind() == MqttBeginExFW.KIND_SUBSCRIBE;
+        final MqttSubscribeBeginExFW mqttSubscribeBeginEx = mqttBeginEx.subscribe();
 
-        final MqttKafkaRouteConfig resolved = binding != null ?
-            binding.resolve(authorization) : null;
+        final MqttKafkaBindingConfig binding = supplyBinding.apply(routedId);
+        final List<String> topicFilters = new ArrayList<>();
+        mqttSubscribeBeginEx.filters().forEach(f -> topicFilters.add(f.pattern().asString()));
 
+        final List<MqttKafkaRouteConfig> routes = binding != null ?
+            binding.resolve(authorization, topicFilters) : null;
```

Comment:
Suggest renaming this method as `resolveAll` indicating it is finding all matches.
Also, suggest passing in `MqttSubscribeBeginExFW` as a parameter instead of building up the topic filter list externally. We may later add support for working with `StringFW` and patterns more directly, so that becomes an implementation detail.